WPMedia/engine-theme-sdk
# TMEDIA-646 upgrade react testing dependencies

## Description
Upgrade react to 17. Make sure tests and previous functionality works

## Jira Ticket
- [PEN-646](https://arcpublishing.atlassian.net/browse/PEN-646)

## Acceptance Criteria
Smoke-test `react-swipeable` after getting a warning from upgrading 

Consider upgrading to get proper react-swipeable support to @6 

All tests should pass 
## Test Steps
- Checkout branch`git checkout TMEDIA-646-upgrade-react-testing-dependencies`
- Clear all node modules and install updates `rm -rf node_modules && npm i`
- Run all tests and working `npm t`
- See storybook cloud or locally `npm run storybook`
- Go to gallery and ensure gallery works as expected 
- Check that you can still swipe as before in the mobile view https://5eed0506faad4f0022fedf95-ovjwcvxobx.chromatic.com/?path=/story/gallery--custom-gallery

Live 
- link engine theme sdk 
- `npx fusion start -f`
- Ensure that the gallery page still works as expected on a page like all-block on mobile http://localhost/pf/test-article-page/?_website=the-gazette

## Effect Of Changes
### Before
- Using react 16, not representing best of engine 

### After
- Using react 17 as expected with engine 

## Dependencies or Side Effects
- the react-swipeable lib inverted its delta for swipes. I adjusted its output based on our implementation 

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
